internal/experiment/generation/environmentvariables.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2021 GramLabs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generation
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	redskyv1beta1 "github.com/thestormforge/optimize-controller/api/v1beta1"
+	"github.com/thestormforge/optimize-controller/internal/scan"
+	"github.com/thestormforge/optimize-controller/internal/sfio"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// EnvironmentVariablesSelector scans for environment variables.
+type EnvironmentVariablesSelector struct {
+	scan.GenericSelector
+	// Regular expression matching the container name.
+	ContainerName string `json:"containerName,omitempty"`
+	// Name of the environment variable to match.
+	VariableName string `json:"variableName,omitempty"`
+	// Path to the environment variable's value.
+	Path string `json:"path,omitempty"`
+	// Prefix that appears before the value.
+	ValuePrefix string `json:"valuePrefix,omitempty"`
+	// Suffix that appears after the value.
+	ValueSuffix string `json:"valueSuffix,omitempty"`
+	// Allowed values for categorical parameters.
+	Values []string `json:"values,omitempty"`
```

Comment:
Should we embed the EnvironmentVariable struct in here?

Reply:
I'm sticking with the approach from the other implementations for now, I would like to switch them all over at once when we expose more of the internals in the app.yaml API.